masonry/src/tree_arena.rs

```diff
@@ -268,6 +268,12 @@ impl<'a, Item> ArenaRefChildren<'a, Item> {
             .map(|child| child.arena_ref(self.id, self.parents_map.parents_map))
     }
 
+    pub fn iter(self) -> impl DoubleEndedIterator<Item = ArenaRef<'a, Item>> + ExactSizeIterator {
+        self.children
+            .iter()
+            .map(move |child| child.arena_ref(self.id, self.parents_map.parents_map))
+    }
```

Comment:
Yeah that's probably a good trade-off and at least avoids quadratic slowdown (which practically results in being completely "stuck" with big enough `N`).

But I think you mean a `HashMap`, i.e. something like `children: HashMap<u64, TreeNode<Item>>`?

Reply:
Right, a hashmap.

Reply:
Ok, should this go into this PR? Or a separate PR, as I think it's somewhat orthogonal?

Reply:
We'll see after #752 merges. Probably a separate PR. Either way, I'd prefer if this PR didn't add an `iter()` method.

Reply:
Sure, as I noted [here](https://github.com/linebender/xilem/pull/752#discussion_r1861927706) (and in zulip) that [this commit](https://github.com/linebender/xilem/pull/753/commits/14d1e6e366e5d834aaca155750b4a52fee9569f0) is temporary (a reason why I tried to isolate these changes, so that I can easily remove it again.

Alright then I'm waiting for the merger of #752 and implement it on top.

Reply:
Ok, good that you mention, otherwise I would've implemented that today or tomorrow.

Reply:
See #772